### PR TITLE
Edge Model Registry - Sidebar and EmptyState

### PR DESCRIPTION
**Pull request by Sara4994**
Closes: https://github.com/opendatahub-io/odh-dashboard/issues/2061

## Description

As part of new Edge specific initiatives, this PR introduces new section for Edge MVP and an empty state component on the new Model Registry page.

Its just the first step in Edge initiative and there will be more features introduced soon.

Screenshot 2023-11-22 at 5 30 01 PM

## How Has This Been Tested?

Click Edge MVP option on the sidebar and select Model Registry

## Test Impact

Integration Testing: Included simple test to check the title of EmptyState has rendered.

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by Gkrumbach07** [comment]
@Sara4994 Make sure you fill out all the template information, including the checkboxes

File: frontend/src/app/AppRoutes.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const AppRoutes: React.FC = () => {
             <Route path="/groupSettings" element={<GroupSettingsPage />} />
           </>
         )}
-
+        <Route path="/modelRegistry" element={<EdgeModelRegistry />} />
```
Comment (Gkrumbach07) [inline frontend/src/app/AppRoutes.tsx:83]: these should all go under a route for `/edge/*`

File: frontend/src/utilities/NavData.tsx [hunk 2]
```diff
@@ -90,6 +90,14 @@ const useDSPipelinesNav = (): NavDataItem[] => {
   ];
 };
 
+const useEdgeMVPNav = (): NavDataItem[] => [
```
Comment (Gkrumbach07) [inline frontend/src/utilities/NavData.tsx:93]: this needs to use feature flags

File: frontend/src/pages/edgeMVP/EdgeModelRegistry.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <Bullseye>
+      <EmptyModelRegistry />
+    </Bullseye>
+  </ApplicationsPage>
+));
+
+EdgeModelRegistryInner.displayName = 'EdgeModelRegistryInner';
+
+const EdgeModelRegistry: React.FC = () => {
+  const { loaded } = useWatchComponents(true);
```
Comment (Gkrumbach07) [inline frontend/src/pages/edgeMVP/EdgeModelRegistry.tsx:22]: why are you using useWatchComponents here?

This should be fetching the created models

**Review by Gkrumbach07 — CHANGES_REQUESTED** [review]

**Comment by Gkrumbach07** [comment]
waiting on tests

**Review by Gkrumbach07 — APPROVED** [review]
/lgtm

@christianvogt can you give the final look at this to merge

for context this is just the feature flag and the first empty state for the edge feature

**Comment by Gkrumbach07** [comment]
waiting on UX as well.

@vconzola This is just for the navbar on the side and one empty state. The wording can change in future PRs if you dont have one now. This is just so we can start putting in a foundation for the architecture of the feature

File: frontend/src/pages/edge/EmptyModelRegistry.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import { EmptyState, EmptyStateBody, EmptyStateIcon, Title } from '@patternfly/react-core';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { Bullseye } from '@patternfly/react-core';
+
+const EmptyModelRegistry: React.FC = () => (
+  <Bullseye>
+    <EmptyState>
+      <EmptyStateIcon icon={PlusCircleIcon} />
+      <Title headingLevel="h2" size="lg" style={{ marginBottom: '10px' }}>
```
Comment (dpanshug) [inline frontend/src/pages/edge/EmptyModelRegistry.tsx:10]:
```suggestion
      <Title headingLevel="h1" size="lg" style={{ marginBottom: '10px' }}>
```

**Review by dpanshug — CHANGES_REQUESTED** [review]

**Review by dpanshug — APPROVED** [review]
/lgtm

**Review by dpanshug — APPROVED** [review]
/lgtm

**Comment by vconzola** [comment]
@Sara4994 After discussions with Gage and the RHOAI dashboard team I've updated the mockups, including the navigation and the empty state.

1. Edge will appear in the navigation above Settings and Resources
2. We'll only have two subheadings under Edge - Models and ML Ops Pipelines (this second one might change)
3. Text on the empty state has been updated. (Still not final, but close enough for now.)

Mockups here: https://docs.google.com/presentation/d/1W9wHeASZsxz4W1UzyN-HfQCqeMhI0dm4IpyxnPBALLo/edit?usp=sharing   Only the mockups with the red dots will be included for December.

cc: @Gkrumbach07

**Comment by Sara4994** [comment]
Thanks @vconzola for providing updated mockups. I have updated the UI as per the suggestions. please have a look if that looks good.

**Comment by dpanshug** [comment]
@Sara4994 Can you update the screenshot in description, easy to review by UX. Also the linting test is failing.

File: frontend/src/utilities/NavData.tsx [hunk 5]
```diff
@@ -90,6 +90,17 @@ const useDSPipelinesNav = (): NavDataItem[] => {
   ];
 };
 
+const useEdgeMVPNav = (): NavDataItem[] =>
+  useAreaCheck(SupportedArea.EDGE, [
+    {
+      id: 'edgemvp',
+      group: { id: 'edge', title: 'Edge MVP' },
+      children: [
+        { id: 'edge-model-registry', label: 'Model registry', href: '/edge/modelRegistry' },
```
Comment (dpanshug) [inline frontend/src/utilities/NavData.tsx:99]:
```suggestion
        { id: 'edge-model-registry', label: 'Models', href: '/edge/modelRegistry' },
```

**Comment by vconzola** [comment]
LGTM.

File: frontend/src/pages/edge/EmptyEdgeModels.tsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+import { PlusCircleIcon } from '@patternfly/react-icons';
+
+const EmptyEdgeModels: React.FC = () => (
+  <PageSection isFilled>
+    <EmptyState>
+      <EmptyStateIcon icon={PlusCircleIcon} />
+      <Title headingLevel="h1" size="lg" style={{ marginBottom: '10px' }}>
+        No models added
+      </Title>
+      <EmptyStateBody style={{ margin: 'auto', width: '50%' }}>
```
Comment (christianvogt) [inline frontend/src/pages/edge/EmptyEdgeModels.tsx:18]: Avoid all custom styles. We must adhere to what PF provides or updates to PF will create inconsistencies in our application.

```suggestion
    <EmptyState variant={EmptyStateVariant.large}>
      <EmptyStateIcon icon={PlusCircleIcon} />
      <Title headingLevel="h1" size="lg">
        No models added
      </Title>
      <EmptyStateBody>
```

**Review by christianvogt — CHANGES_REQUESTED** [review]

**Review by christianvogt — APPROVED** [review]

**Pull request merged**